### PR TITLE
add re-regsiter and delete options to service catalog

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed eclipse-xpanse/xpanse#1655

image
image
image
image
image

### CODE REVIEW

File: src/components/content/catalog/services/re-register/reRegisterService.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                onCancel={() => {
+                    reRegister();
+                }}
+            >
+                <Button
+                    icon={<CloseCircleOutlined />}
+                    type='primary'
+                    className={catalogStyles.catalogManageBtnClass}
+                    disabled={
+                        serviceRegistrationState !== ServiceTemplateDetailVo.serviceRegistrationState.UNREGISTERED
```
Comment (swaroopar) [inline src/components/content/catalog/services/re-register/reRegisterService.tsx:43]: in the screenshots, i see that the button is enabled even after the re-register request is submitted. The button must be disabled, if the re-register is submitted.

File: src/components/content/catalog/services/delete/DeleteService.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <Popconfirm
+                title='Delete the service'
+                description='Are you sure to delete this service?'
+                cancelText='Yes'
+                okText='No'
+                onCancel={() => {
+                    deleteService();
+                }}
+            >
+                <Button
```
Comment (swaroopar) [inline src/components/content/catalog/services/delete/DeleteService.tsx:38]: same here. the delete button is enabled even after the delete is successful.

File: src/components/content/catalog/services/delete/DeleteResult.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    };
+
+    if (
+        deleteRequestState[0] &&
+        serviceRegistrationState === ServiceTemplateDetailVo.serviceRegistrationState.UNREGISTERED
+    ) {
+        if (deleteRequestState[0].status === 'success') {
+            return (
+                <Alert
+                    message='Service Delete Successfully'
```
Comment (swaroopar) [inline src/components/content/catalog/services/delete/DeleteResult.tsx:43]: should be `Service Deleted Successfully`

File: src/components/content/catalog/services/delete/DeleteResult.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                    description={'Service Delete successfully.'}
```
Comment (swaroopar) [inline src/components/content/catalog/services/delete/DeleteResult.tsx:44]: description can be something else - Service removed from the database completely.

File: src/components/content/catalog/services/re-register/reRegisterResult.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    };
+
+    if (
+        useReRegisterRequestState[0] &&
+        serviceRegistrationState === ServiceTemplateDetailVo.serviceRegistrationState.UNREGISTERED
+    ) {
+        if (useReRegisterRequestState[0].status === 'success') {
+            return (
+                <Alert
+                    message='Service Re-Register Successfully'
```
Comment (swaroopar) [inline src/components/content/catalog/services/re-register/reRegisterResult.tsx:43]: Service re-registered successfully

File: src/components/content/catalog/services/re-register/reRegisterResult.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                    description={'Service submitted for review successfully.'}
```
Comment (swaroopar) [inline src/components/content/catalog/services/re-register/reRegisterResult.tsx:44]: Service submitted for review of cloud provider.

File: src/components/content/catalog/services/unregister/UnregisterResult.tsx [hunk 7]
```diff
@@ -31,12 +33,15 @@ export function UnregisterResult({
...
+    if (
+        useUnregisterRequestState[0] &&
+        serviceRegistrationState !== ServiceTemplateDetailVo.serviceRegistrationState.UNREGISTERED
+    ) {
         if (useUnregisterRequestState[0].status === 'success') {
             return (
                 <Alert
                     message='Service Unregistered Successfully'
-                    description={'Service removed from the catalog.'}
+                    description={'Service removed from the catalog successfully.'}
```
Comment (swaroopar) [inline src/components/content/catalog/services/unregister/UnregisterResult.tsx:44]: the earlier description was fine. No need to change that

File: src/components/content/catalog/services/unregister/UnregisterResult.tsx [hunk 8]
```diff
@@ -31,12 +33,15 @@ export function UnregisterResult({
         });
     };
 
-    if (useUnregisterRequestState[0]) {
+    if (
```
Comment (swaroopar) [inline src/components/content/catalog/services/unregister/UnregisterResult.tsx:36]: is it necessary to change this if condition ?
It must work just with the old condition.

File: src/components/content/catalog/services/delete/DeleteResult.tsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    const onRemove = () => {
+        void queryClient.refetchQueries({ queryKey: getQueryKey(category) });
+        navigate({
+            pathname: catalogPageRoute,
+            hash: '#' + category,
+        });
+    };
+
+    if (
```
Comment (swaroopar) [inline src/components/content/catalog/services/delete/DeleteResult.tsx:36]: should we enough just to check the deleteRequestState.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Comment by Alice1319** [comment]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**